### PR TITLE
TW-1578: Improve download attachment on web

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket

- #1578 

### Resolved

- Download files in non-encrypted chat

https://github.com/linagora/twake-on-matrix/assets/99852347/aa3e0ed2-fe98-4fb8-b142-0c8fb0434470

- download files in encrypted chat

https://github.com/linagora/twake-on-matrix/assets/99852347/351b9562-2f06-4593-adaa-9efc87f68d25

### CODE REVIEW

File: lib/data/network/media/media_api.dart [hunk 1]
```diff
@@ -72,6 +73,31 @@ class MediaAPI {
     );
   }
 
+  Future<Uint8List> downloadAttachmentWeb({
```
Comment (hoangdat) [inline lib/data/network/media/media_api.dart:76]:
```suggestion
  Future<Uint8List> downloadFileWeb({
```

File: lib/utils/matrix_sdk_extensions/download_file_extension.dart [hunk 2]
```diff
@@ -286,6 +287,193 @@ extension DownloadFileExtension on Event {
     );
   }
 
+  Future<MatrixFile?> downloadAttachmentWeb({
```
Comment (hoangdat) [inline lib/utils/matrix_sdk_extensions/download_file_extension.dart:290]: should we separate extension for web?

**Comment by hoangdat** [comment]
please demo:
- [ ] download files in non-encrypted chat
- [ ] download files in encrypted chat
- [ ] download file which have thumbnail in non-encrypted chat
- [ ] download file which have thumbnail in encrypted chat

File: lib/utils/matrix_sdk_extensions/download_file_web_extension.dart [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+extension DownloadFileWebExtension on Event {
+  Future<MatrixFile?> downloadFileWeb({
+    required StreamController<Either<Failure, Success>>
+        downloadStreamController,
+    CancelToken? cancelToken,
+  }) async {
+    if (!canContainAttachment()) {
+      throw (
+        "downloadFileWeb: This event has the type '$type' and so it can't contain an attachment.",
+      );
```
Comment (sherlockvn) [inline lib/utils/matrix_sdk_extensions/download_file_web_extension.dart:24]: make an exception

File: lib/utils/matrix_sdk_extensions/download_file_web_extension.dart [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      streamController.add(
+        Left(
+          DownloadFileFailureState(exception: e),
+        ),
+      );
+    }
+  }
+
+  Future<Uint8List?> _decryptAttachmentWeb({
```
Comment (sherlockvn) [inline lib/utils/matrix_sdk_extensions/download_file_web_extension.dart:172]: you should put into 1 method, and put it in sdk

**Review by hoangdat — APPROVED** [review]

**Pull request merged**